### PR TITLE
Coerce empty string to null

### PR DESCRIPTION
**Pull request by shmax**
Fix for the null coercion portion of https://github.com/justinrainbow/json-schema/issues/379

Coerce empty sting to null for type "null".

@erayd @bighappyface @gtuccini

### CODE REVIEW

File: src/JsonSchema/Constraints/TypeConstraint.php [hunk 1]
```diff
@@ -208,6 +208,9 @@ protected function validateType(&$value, $type)
         }
 
         if ('null' === $type) {
+            if ($coerce) {
+                $value = $this->toNull($value);
+            }
             return is_null($value);
```
Comment (erayd) [inline src/JsonSchema/Constraints/TypeConstraint.php]: Need a newline above `return is_null($value);` to make the style checker happy.

File: tests/Constraints/CoerciveTest.php [hunk 2]
```diff
@@ -128,7 +128,7 @@ public function getValidCoerceTests()
                   "boolean":"true",
                   "object":{},
                   "array":[],
-                  "null":null,
+                  "null":"",
```
Comment (erayd) [inline tests/Constraints/CoerciveTest.php]: I notice that you have lots of explicit tests in `testValidCoerceCases`, in addition to simply requiring the schema to validate, but that null coercion hasn't been added as one of them. Is there a reason for that?

Comment (shmax) [inline tests/Constraints/CoerciveTest.php]: I don't understand. Can you provide an example? 

I believe I originally created this file by duplicating BasicTypesTest.

Comment (shmax) [inline tests/Constraints/CoerciveTest.php]: Ohhh, I think I get what you mean. One sec...

File: src/JsonSchema/Constraints/TypeConstraint.php [hunk 3]
```diff
@@ -211,6 +211,7 @@ protected function validateType(&$value, $type)
             if ($coerce) {
                 $value = $this->toNull($value);
             }
+
```
Comment (erayd) [inline src/JsonSchema/Constraints/TypeConstraint.php]: The newline is good, but need to remove the unneeded whitespace on this line.

File: tests/Constraints/CoerciveTest.php (hunk 2 above)
Comment (shmax) [inline tests/Constraints/CoerciveTest.php]: Added a post-validate test for null.

I also removed what looked like a duplicated block of explicit tests in the same section of code. Not sure if there was any purpose for it...

**Comment by erayd** [comment]
I think we're good - thanks for doing this PR :-).

**Comment by erayd** [comment]
Is this PR still necessary? Seems to me that #384 has made it redundant.

**Comment by shmax** [comment]
Right you are. Closing...

**Pull request closed without merging**